### PR TITLE
glibc: Forbid installation if your system glibc is newer

### PR DESCRIPTION
**Pull request by rwhogg**
Formulae built from source against a newer Glibc
may crash after the installation, as they're looking
for symbols not provided by the older library.

Signed-off-by: Bob W. Hogg <rwhogg@linux.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by rwhogg** [comment]
Successful except for keep-old on Circle. (But we're not pulling the bottle for this PR anyways)

File: Formula/glibc.rb [hunk 1]
```diff
@@ -36,6 +36,25 @@ def message
   end
 end
 
+class NotOlderGlibcRequirement < Requirement
+  fatal true
+
+  def system_version
+    @system_glibc_version ||= GlibcRequirement.system_version
```
Comment (sjackman) [inline Formula/glibc.rb]: `GlibcRequirement.system_version` caches its result so there's no need to duplicate that work here.

File: Formula/glibc.rb [hunk 2]
```diff
@@ -36,6 +36,25 @@ def message
   end
 end
 
+class NotOlderGlibcRequirement < Requirement
```
Comment (sjackman) [inline Formula/glibc.rb]: Can you please rename this to `BrewedGlibcNotOlderRequirement` ?

File: Formula/glibc.rb [hunk 3]
```diff
@@ -36,6 +36,25 @@ def message
...
+  end
+
+  satisfy(:build_env => false) do
+    system_version <= Glibc.version
+  end
+
+  def message
+    <<-EOS.undent
+      Installing an older version of glibc than your system version can break formulae installed from source.
+      Your system has glibc version #{system_version}.
```
Comment (sjackman) [inline Formula/glibc.rb]: Can you please reword this to…
```
Your system's glibc version is foo, and Linuxbrew's gcc version is bar.
Installing a version of glibc that is older than your system's can break formulae installed from source.
```

**Comment by maxim-belkin** [comment]
Circle succeeded except for the `--keep-old` thing (as usual):
```
./glibc-2.19.x86_64_linux.bottle.tar.gz
  bottle do
    sha256 "c73f6f504b3bbbf219ad4c8f673e3386117190216172fe23242692cdeefb7b78" => :x86_64_linux
  end
==> brew bottle --merge --write --no-commit ./glibc-2.19.x86_64_linux.bottle.json --keep-old
Error: --keep-old was passed but there are changes in:
sha256 => x86_64_linux
Error: Kernel.exit
==> glibc
==> FAILED
```
